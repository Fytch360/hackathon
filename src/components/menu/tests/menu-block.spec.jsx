import React from 'react'
import { shallow } from 'enzyme'

import { MenuBlock, MenuItem } from '..'

describe('<MenuBlock />', () => {
    test('is available', () => {
        expect(MenuBlock).toBeDefined()
    })

    test('Snapshot', () => {
        const wrapper = shallow(
            <MenuBlock a11y={{ title: 'hmm' }}>
                <MenuItem />
            </MenuBlock>
        )

        expect(wrapper).toMatchSnapshot()
    })

    test('Should throw onClose for children', () => {
        const onCloseMock = jest.fn()
        const wrapper = shallow(
            <MenuBlock a11y={{ title: 'HMM' }} onClose={onCloseMock}>
                <MenuItem />
                <MenuItem />
                <MenuItem />
            </MenuBlock>
        )

        wrapper.find('ul').children().map((child) => expect(child.props().onClose).toBe(onCloseMock))
        
    })

    test('Should throw activeDescendant for children', () => {
        const activeDescendantValue = 'Important value btw'
        const wrapper = shallow(
            <MenuBlock a11y={{ title: 'Hmmmm...' }} activeDescendant={activeDescendantValue}>
                <MenuItem />
                <MenuItem />
                <MenuItem />
            </MenuBlock>
        )

        wrapper.find('ul').children().map((child) => expect(child.props().activeDescendant).toBe(activeDescendantValue))
    })
})
