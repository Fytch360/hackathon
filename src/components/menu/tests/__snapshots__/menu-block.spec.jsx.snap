// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MenuBlock /> Snapshot 1`] = `
<li>
  <BlockStyled
    aria-label="hmm"
  >
    <MenuItem
      key=".0"
    />
  </BlockStyled>
</li>
`;
